src/package.cc

```diff
@@ -0,0 +1,32 @@
+#include "package.h"
+#include "error.h"
+
+namespace cyclus {
+
+int Package::next_id_ = 1;
+
+Package::Ptr Package::Create() {
+  Ptr p(new Package());
+  return p;
+}
+
+Package::Ptr Package::Create(std::string name, double fill_min, double fill_max, std::string strategy) {
+  if (fill_min < 0 || fill_max < 0) {
+    throw ValueError("fill_min and fill_max must be non-negative");
+  }
+  else if (fill_min > fill_max) {
+    throw ValueError("fill_min must be less than or equal to fill_max");
+  }
+  Ptr p(new Package(name, fill_min, fill_max, strategy));
+  return p;
+}
+
+int Package::id() {
+  return id_;
+}
+  
+Package::Package() : id_(next_id_++), fill_min_(0), fill_max_(std::numeric_limits<double>::max()), strategy_("first") {}
```

Comment:
What does `strategy_ == "first"` imply/mean?  Should this just be blank?  Also, shouldn't you initialize the `name` as something, including an empty string?

Reply:
This is me starting to think about the fill strategies. For now, just planning to do fill first `first` and fill equally `equal`. Fill first would be the default. But it's not set in stone, happy to use another strategy